### PR TITLE
Optimize and improve built-in PNG writer

### PR DESCRIPTION
**Pull request by sfan5**
both in performance and in efficiency of the resulting file

example size reduction on a grayscale image:
　`testnodes_generated_ck.png` 5049 -> 2025 (-59%)
savings on RGB images are usually less good

## To do

This PR is  Ready for Review.

## How to test

1. check devtest unittests
2. run `file games/devtest/mods/testnodes/textures/testnodes_generated_*`

### CODE REVIEW

**Comment by SmallJoker** [comment]
@ryvnf On server builds there is yet only a small subset of dependencies. I suppose that was the reason why this code was written in the first place. Of course it would be a saner solution to rely on libraries provided by the repository (that are likely already installed) for server builds as well.

**Comment by sfan5** [comment]
The server currently only depends on `curl`, `sqlite3`, `luajit`, `leveldb`, `zlib` and `zstd` (of which some are optional) plus the IrrlichtMt *headers*.
We're very careful with adding dependencies, which is why the PNG writing was NIH'd.

That said this wariness might not be justified and once we add PNG reading support we'll need libpng anyway in which case this code should be replaced.
I also don't have plans to add more color types, 16-bit or similarly advanced stuff. The only thing that'd make sense is palette support.

**Comment by sfan5** [comment]
I quickly investigated the libpng API and it looks like it does not provide functions to down-convert the color prior to writing.
So the most substantial part (of the new code), which is only really 35 lines, would have to stay anyway.

File: games/devtest/mods/testnodes/textures.lua [hunk 1]
```diff
@@ -129,17 +142,21 @@ for i=1, #fractal do
 		b = floor(abs(1 - fractal[i]) * 255),
 		a = 255,
 	}
-	data_ck[i] = checker[i] > 0 and "#F80" or "#000"
+	data_ck[i] = checker[i] > 0 and "#888" or "#000"
 end
 
+fractal = nil
+frac_emb = nil
+checker = nil
```
Comment (SmallJoker) [inline games/devtest/mods/testnodes/textures.lua:150]: You could put a `do` + `end` block around the relevant code to automatically let the variables run out of scope. Diff viewers can ignore whitespace changes.

**Review by SmallJoker — APPROVED** [review]
The devtest startup code works, thus LGTM.
The comment may or may not be addressed for this approval.

File: games/devtest/mods/testnodes/textures.lua (hunk 1 above)
Comment (sfan5) [reply games/devtest/mods/testnodes/textures.lua:150]: it's a bunch of lines so meh

**Pull request merged**